leaderelection/src/main/java/io/scalecube/leaderelection/HeartbeatScheduler.java

```diff
@@ -0,0 +1,51 @@
+package io.scalecube.leaderelection;
+
+import io.scalecube.cluster.ICluster;
+import io.scalecube.transport.Message;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/** 
+ * heartbeat scheduler broadcast heartbeats on the cluster to maintain leadership members in the cluster 
+ * expects this heartbeat in case heartbeat does not reach members within X time leader election process 
+ * is initiated. 
+ * 
+ * @author Ronen Nachmias
+ */
+
+public class HeartbeatScheduler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HeartbeatScheduler.class);
+  private final ScheduledThreadPoolExecutor scheduler = new ScheduledThreadPoolExecutor(1);

```

Comment:
Is it really required to have dedicated thread pool for this heartbeat scheduler?
Can we use some shared resource here?